### DISCUSSION TITLE
'lit-localize can only be configured once'

### DISCUSSION THREAD

**Discussion by bobby-jordan** [lit/lit#3366] [Q&A]
Hello everybody,

I am working on 2 independently deployed web components, nested in one another.
Both components have:
- Installed [@lit/localize and @lit/localize-tools.](https://lit.dev/docs/localization/overview/#installation)
- Use [Runtime localization mode](https://lit.dev/docs/localization/runtime-mode/)
- The localization is [Automatically re-rendered ](https://lit.dev/docs/localization/runtime-mode/#automatically-re-render)
- The approach for loading the local modules is [lazy-loading](https://lit.dev/docs/localization/runtime-mode/#approaches-for-loading-locale-modules)

![image](https://user-images.githubusercontent.com/67242163/195707052-f21ade08-82e9-40be-9442-f16d6ac333bf.png)

Independently they work being packaged, but when I install the child-web-component in the parent-web-component, I am getting an error message: **lit-localize can only be configured once** 
![image](https://user-images.githubusercontent.com/67242163/195706591-0f9a43da-54a6-4b6a-bbbc-687f4ae8cc51.png)
![image](https://user-images.githubusercontent.com/67242163/195706624-110bd2c0-d03d-4b2a-abf1-d9aad4be3757.png)
![image](https://user-images.githubusercontent.com/67242163/195706636-b0d48498-4b7b-44c8-914e-7ca442342a02.png)
![image](https://user-images.githubusercontent.com/67242163/195706656-7ba1fac7-6b04-42ba-8f0f-e947dbd821fc.png)

(reference file here: [https://github.com/lit/lit/blob/main/packages/localize/src/lit-localize.ts](https://github.com/lit/lit/blob/main/packages/localize/src/lit-localize.ts))

My question is: how can I use lit-localization in a consistent matter in both web components?

**Comment by aomarks** [comment]
Unfortunately this use case does not currently have good support in Lit Localize. Improved support is planned in https://github.com/lit/lit/issues/1661. In particular see this comment, where an outline of the proposed new API is shown: https://github.com/lit/lit/issues/1661#issuecomment-814430658

The intention for `configureLocalization` is that it is only ever called **once per application**. A component library should never itself call `configureLocalization`. However, if one of your dependencies is installed via `node_modules`, then the problem with this is that `localize-tools` doesn't automatically scan inside of your `node_modules` dependencies for `msg` calls, so those translations won't get generated/loaded.

One pattern that could work as a workaround would be to explicitly specify the source files of both components from a single `lit-localize.json` file (i.e. listing both in the `inputFiles` option https://github.com/lit/lit/issues/1661#issuecomment-814430658, where one of them is e.g. `./node_modules/other-component/**/*.js`).

I hope that helps, apologies for the complexity, things will be simpler when  https://github.com/lit/lit/issues/1661 is implemented.
